test/e2e/kubectl/kubectl.go

```diff
@@ -951,12 +953,14 @@ metadata:
 
 		ginkgo.It("should create/apply a valid CR with arbitrary-extra properties for CRD with partially-specified validation schema", func() {
```

Comment:
did you run this test? I expected to have to set `x-kubernetes-preserve-unknown-fields` in the schema for this

Reply:
I've added it in the schema root (https://github.com/kubernetes/kubernetes/pull/81497/files#diff-6b2122024cc63dc0fd467f3320a9005eR139) which is used in 7 places. I'd like to minimize use of `x-kubernetes-preserve-unknown-fields` in our e2e test to make them a better example to users. Okay as a follow up?

Reply:
> I've added it in the schema root 

aren't you overwriting that in this test with this line: `crd.Spec.Versions[i].Schema = &apiextensionsv1.CustomResourceValidation{OpenAPIV3Schema: props}`

> I'd like to minimize use of x-kubernetes-preserve-unknown-fields in our e2e test to make them a better example to users

The point of this test specifically is to test compatibility of kubectl with a partially specified schema. I think this needs to specify `x-kubernetes-preserve-unknown-fields: true` at the schema root to do that.

Reply:
(I'm also surprised `kubectl apply` did not complain about applying the extra fields with a full schema published)

Reply:
figured it out... the remnants of https://github.com/kubernetes/kubernetes/pull/78770 tolerated kubectl failure if a schema with properties was published (which it should not have... the schema publishing in the partial-schema+tolerate-unknown-fields case was fixed in https://github.com/kubernetes/kubernetes/pull/79636 but this e2e test still incorrectly tolerated kubectl failure)

Reply:
Fixed by removing `expectSuccess` conditions and adding `XPreserveUnknownFields` to root of schema.

Success case: https://gist.github.com/jpbetz/fec17da9fb3fd17bfb5a76e36ef95cea

I'm also validating that without `XPreserveUnknownFields` that the test (correctly) fails. Will report back if that is not the case.

Reply:
Without `XPreserveUnknownFields` got the failure we expect: `"unknown field "extraProperty"`:

https://gist.github.com/jpbetz/00484e16513801f981a81d7e77d9eae6